extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -26,8 +33,11 @@ public function initEmail() {
 
 		$this->title = \Title::newFromText( $titleText, $titleNamespace );
 		$this->summary = $this->getVal( 'summary' );
-		$this->currentRevId = $this->getVal( 'currentRevId' );
-		$this->previousRevId = $this->getVal( 'previousRevId' );
+
+		if ( $this->title instanceof \Title ) {
+			$this->currentRevId = $this->title->getLatestRevID( \Title::GAID_FOR_UPDATE );
+			$this->previousRevId = $this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE );
+		}
```

Comment:
```recentchanges``` table does not contain any value for ```*RevId```:
````
select * from recentchanges where rc_log_action in ("delete", "move", "protect") order by rc_id desc limit 20;
+--------+----------------+----------------+---------+---------------+--------------+-----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+----------+--------+--------+-----------+---------------+---------------+---------+----------------+-------------------+--------------+-----------------+------------+------------+------------+----------+-------------+---------------+----------------------------------------------------------------------------------------------------------+
| rc_id  | rc_timestamp   | rc_cur_time    | rc_user | rc_user_text  | rc_namespace | rc_title                                                  | rc_comment                                                                                                                        | rc_minor | rc_bot | rc_new | rc_cur_id | rc_this_oldid | rc_last_oldid | rc_type | rc_moved_to_ns | rc_moved_to_title | rc_patrolled | rc_ip           | rc_old_len | rc_new_len | rc_deleted | rc_logid | rc_log_type | rc_log_action | rc_params                                                                                                |
+--------+----------------+----------------+---------+---------------+--------------+-----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+----------+--------+--------+-----------+---------------+---------------+---------+----------------+-------------------+--------------+-----------------+------------+------------+------------+----------+-------------+---------------+----------------------------------------------------------------------------------------------------------+
| 796576 | 20150202205745 | 20150202205745 | 5654073 | James test    |            0 | MuppetTestPage                                            |                                                                                                                                   |        0 |      0 |      0 |    189660 |             0 |             0 |       3 |              0 |                   |            1 | 65.19.148.1     |       NULL |       NULL |          0 |   183277 | move        | move          | a:2:{s:9:"4::target";s:18:"MuppetTestPageCopy";s:10:"5::noredir";s:1:"0";}                               |
| 796434 | 20140908194629 | 20140908194629 | 1342530 | QATestsStaff  |            6 | The_JET_Bicycle_-_The_most_dangerous_unsafe_bike_EVER     |                                                                                                                                   |        0 |      0 |      0 |    189627 |             0 |             0 |       3 |              0 |                   |            1 | 65.19.148.1     |       NULL |       NULL |          0 |   183232 | delete      | delete        | a:0:{}                                                                                                   |
| 796020 | 20140805182355 | 20140805182355 |  455627 | Jbrangwynne53 |            6 | Punch_out.jpg                                             |                                                                                                                                   |        0 |      0 |      0 |    189532 |             0 |             0 |       3 |              0 |                   |            1 | 72.224.248.239  |       NULL |       NULL |          0 |   183134 | delete      | delete        | a:0:{}                                                                                                   |
| 795959 | 20140805103522 | 20140805103522 |   33935 | H rytter      |            0 | Once_more                                                 | spam                                                                                                                             |        0 |      0 |      0 |    189519 |             0 |             0 |       3 |              0 |                   |            1 | 188.176.19.239  |       NULL |       NULL |          0 |   183120 | delete      | delete        | a:0:{}                                                                                                   |
| 795785 | 20140804183227 | 20140804183227 |   10637 | Scarecroe     |            6 | TwiddleDream.jpg                                          |                                                                                                                                   |        0 |      0 |      0 |         0 |             0 |             0 |       3 |              0 |                   |            1 | 70.192.20.137   |       NULL |       NULL |          0 |   183103 | move        | move          | a:2:{s:9:"4::target";s:25:"File:TwiddleDreambook.jpg";s:10:"5::noredir";s:1:"1";}                        |
| 795767 | 20140804163618 | 20140804163618 |   12385 | Aleal         |            0 | The_Muppets_(2011)_books                                  | broken link, only used on an archived talk page                                                                                  |        0 |      0 |      0 |    142511 |             0 |             0 |       3 |              0 |                   |            1 | 173.175.171.147 |       NULL |       NULL |          0 |   183097 | delete      | delete        | a:0:{}                                                                                                   |
| 795738 | 20140804044534 | 20140804044534 |   12385 | Aleal         |            6 | Carroll_Spinney_Interview_Part_2_of_6_-_EMMYTVLEGENDS.ORG | broken redirect                                                                                                                  |        0 |      0 |      0 |    174336 |             0 |             0 |       3 |              0 |                   |            1 | 173.175.171.147 |       NULL |       NULL |          0 |   183093 | delete      | delete        | a:0:{}                                                                                                   |
| 795720 | 20140804041638 | 20140804041638 |   10370 | Toughpigs     |            0 | Muppet_cell_phone_cases_(Mobile_Style_Accessories)        |                                                                                                                                   |        0 |      0 |      0 |    189478 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183083 | move        | move          | a:2:{s:9:"4::target";s:45:"Muppet phone cases (Mobile Style Accessories)";s:10:"5::noredir";s:1:"0";}    |
| 795671 | 20140804023901 | 20140804023901 |   10370 | Toughpigs     |            6 | Bioworld_animal_laplander.jpg                             |                                                                                                                                   |        0 |      0 |      0 |    189446 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183063 | delete      | delete        | a:0:{}                                                                                                   |
| 795670 | 20140804023900 | 20140804023900 |   10370 | Toughpigs     |            6 | Bioworld_beaker_laplander_2.jpg                           |                                                                                                                                   |        0 |      0 |      0 |    189455 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183062 | delete      | delete        | a:0:{}                                                                                                   |
| 795669 | 20140804023859 | 20140804023859 |   10370 | Toughpigs     |            6 | Bioworld_beaker_laplander.jpg                             |                                                                                                                                   |        0 |      0 |      0 |    189449 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183061 | delete      | delete        | a:0:{}                                                                                                   |
| 795668 | 20140804023858 | 20140804023858 |   10370 | Toughpigs     |            6 | Bioworld_kermit_peruvian.jpg                              |                                                                                                                                   |        0 |      0 |      0 |    189453 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183060 | delete      | delete        | a:0:{}                                                                                                   |
| 795667 | 20140804023845 | 20140804023845 |   10370 | Toughpigs     |            6 | Bioworld_kermit_peruvian_2.jpg                            |                                                                                                                                   |        0 |      0 |      0 |    189456 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183059 | delete      | delete        | a:0:{}                                                                                                   |
| 795646 | 20140804021332 | 20140804021332 |   10370 | Toughpigs     |            6 | Last_slumber_party_3.jpg                                  |                                                                                                                                   |        0 |      0 |      0 |    189415 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183042 | delete      | delete        | a:0:{}                                                                                                   |
| 795634 | 20140804001258 | 20140804001258 |   10370 | Toughpigs     |            0 | Muppet_gloves_(Hot_Topic)                                 |                                                                                                                                   |        0 |      0 |      0 |         0 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183037 | move        | move          | a:2:{s:9:"4::target";s:27:"Muppet gloves (Concept One)";s:10:"5::noredir";s:1:"1";}                      |
| 795632 | 20140804000655 | 20140804000655 |   10370 | Toughpigs     |            0 | Muppet_knit_hats                                          |                                                                                                                                   |        0 |      0 |      0 |    189438 |             0 |             0 |       3 |              0 |                   |            1 | 98.210.22.193   |       NULL |       NULL |          0 |   183036 | move        | move          | a:2:{s:9:"4::target";s:26:"Muppet hats (Disney Parks)";s:10:"5::noredir";s:1:"0";}                       |
| 795581 | 20140803215225 | 20140803215225 |   22439 | Wikia         |         1201 | The_Boss_of_You/@comment-Wikia-20140627212747             | Automatically moved page while renaming the user "[[User:The Boss of You|The Boss of You]]" to "[[User:Mabel Pines|Mabel Pines]]" |        0 |      1 |      0 |    189399 |             0 |             0 |       3 |              0 |                   |            1 | 127.0.0.1       |       NULL |       NULL |          0 |   182997 | move        | move          | a:2:{s:9:"4::target";s:48:"Thread:Mabel Pines/@comment-Wikia-20140627212747";s:10:"5::noredir";s:1:"0";} |
| 795580 | 20140803215224 | 20140803215224 |   22439 | Wikia         |         1200 | The_Boss_of_You                                           | Automatically moved page while renaming the user "[[User:The Boss of You|The Boss of You]]" to "[[User:Mabel Pines|Mabel Pines]]" |        0 |      1 |      0 |    189398 |             0 |             0 |       3 |              0 |                   |            1 | 127.0.0.1       |       NULL |       NULL |          0 |   182996 | move        | move          | a:2:{s:9:"4::target";s:24:"Message Wall:Mabel Pines";s:10:"5::noredir";s:1:"0";}                         |
| 795579 | 20140803215224 | 20140803215224 |   22439 | Wikia         |            2 | The_Boss_of_You                                           | Automatically moved page while renaming the user "[[User:The Boss of You|The Boss of You]]" to "[[User:Mabel Pines|Mabel Pines]]" |        0 |      1 |      0 |    189397 |             0 |             0 |       3 |              0 |                   |            1 | 127.0.0.1       |       NULL |       NULL |          0 |   182995 | move        | move          | a:2:{s:9:"4::target";s:16:"User:Mabel Pines";s:10:"5::noredir";s:1:"0";}                                 |
| 795414 | 20140802205923 | 20140802205923 | 1070100 | Oscarfan      |            6 | BusStopSign.jpg                                           |                                                                                                                                   |        0 |      0 |      0 |         0 |             0 |             0 |       3 |              0 |                   |            1 | 69.115.174.107  |       NULL |       NULL |          0 |   182969 | move        | move          | a:2:{s:9:"4::target";s:25:"File:AtTheBusStopSign.jpg";s:10:"5::noredir";s:1:"1";}                        |
+--------+----------------+----------------+---------+---------------+--------------+-----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------+----------+--------+--------+-----------+---------------+---------------+---------+----------------+-------------------+--------------+-----------------+------------+------------+------------+----------+-------------+---------------+----------------------------------------------------------------------------------------------------------+
20 rows in set (0.08 sec)
````

https://github.com/Wikia/app/blob/dev/includes/RecentChange.php#L246-L247
I understand, that it can cause raise conditions, on the other hand handling edits separately means that we make email flow more complex.

Do we have any numbers about multiple edits on the same article in short time of period (lets say <5s)?